### ISSUE TITLE
Public Hosted Zone  using troposphere 

### ISSUE THREAD

**Issue by naveenpolamreddi** [cloudtools/troposphere#980]
Can we create a public hosted zone using troposphere..? If so can you please give me a snippet on how to do that where the hosted zone should log on the query calls to the cloudwatch log group. When i am writing the below statement it errs out. Can you please suggest where I am going wrong here.

HostedZones = template.add_resource(HostedZone(HostedZoneConfig={"Comment":"Public DNS Domain"}, Name="aws.kloudlabs.in",QueryLoggingConfig= {'CloudWatchLogsLogGroupArn':'arn:aws:logs:us-east-1:123456789:log-group:np-aws-kloudlabs:*'}))

**Comment by naveenpolamreddi** [comment]
HostedZones = template.add_resource(HostedZone(HostedZoneConfig={"Comment":"Public DNS Domain"}, Name="aws.kloudlabs.in",QueryLoggingConfig={'CloudWatchLogsLogGroupArn':'arn:aws:logs:us-east-1:1234456789:log-group:np-aws-publicdns:*'}))                                        Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() takes at least 2 arguments (1 given)

**Comment by markpeek** [comment]
Likely you would want to structure your code like this below (runs via troposphere but untested against AWS). The TypeError you are getting is due to name specifying the title ("zone" in this case) for the HostedZone resource.
```
from troposphere import Template
from troposphere.route53 import (
    HostedZone, HostedZoneConfiguration, QueryLoggingConfig
)

t = Template()
HostedZones = t.add_resource(HostedZone(
    "zone",
    HostedZoneConfig=HostedZoneConfiguration(Comment="Public DNS Domain"),
    Name="aws.kloudlabs.in",
    QueryLoggingConfig=QueryLoggingConfig(
        CloudWatchLogsLogGroupArn='arn:aws:logs:us-east-1:1234456789:log-group:np-aws-publicdns:*'
    )
))```

**Comment by naveenpolamreddi** [comment]
Thanks Mark for responding fast. So Creating public DNS thru troposphere is untested ?

**Comment by markpeek** [comment]
No, I'm not saying it is untested through troposphere. I'm sure there are quite a few stacks created using troposphere that are creating public DNS. I meant the above snippet of code was not run through CloudFormation and I was just posting out an example that *should* work.

**Comment by naveenpolamreddi** [comment]
HI Mark,

  Thanks for your help, Can you please point me to the code snippet which helps to create a Public hostedzone with logging enabled to cloudwatch log group. I am trying to deploy a stack through troposphere and the cloudformation stack is failing with the below error.

Error Message: The resource policy that you're using for Route 53 query logging doesn't grant Route 53 sufficient permission to create a log stream in the specified log group.

**Comment by naveenpolamreddi** [comment]
This is the code snippet that i have written, 

https://pastebin.com/raw/8Sb6iXss

**Comment by naveenpolamreddi** [comment]
I have managed to resolve it. You need to set resource policy on the cloudwatch loggroup before you can deploy a public hostedzone that logs to the cloudwatch group. So you need to generate two different templtes one for cloudwatch log group and another template for public hosted zone.
